subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultCopySpecCodec.kt

```diff
@@ -54,8 +58,8 @@ class DefaultCopySpecCodec(
             writeBoolean(value.includeEmptyDirs)
             writeBoolean(value.isCaseSensitive)
             writeString(value.filteringCharset)
-            writeNullableSmallInt(value.dirMode)
-            writeNullableSmallInt(value.fileMode)
+            writeNullableSmallInt(value.dirPermissions.map { it.toUnixNumeric() }.orNull)
+            writeNullableSmallInt(value.filePermissions.map { it.toUnixNumeric() }.orNull)
```

Comment:
Definitely not.

Reply:
Why not?

Reply:
Because what you suggest is about permissions for ONE SPECIFIC class of user and what we need here is all permissions for all classes of users.

Reply:
Maybe we are misunderstanding each other. What I mean is it would be better to use a method handle instead of a closure. Is there any reason not to do that?

Reply:
Ahh, that definitely is ok. The class name is wrong though in the suggestion, that's what I was arguing against. Lost in translation, will fix.